src/main/java/org/testng/GuiceHelper.java

```diff
@@ -33,15 +40,41 @@ Injector getInjector(IClass iClass) {
     List<Module> moduleInstances =
         Lists.newArrayList(getModules(guice, parentInjector, iClass.getRealClass()));
 
-    // Reuse the previous injector, if any
+    if (isStringNotEmpty(context.getSuite().getParentModule())) {
+      Class<Module> parentModule = (Class<Module>) ClassHelper.forName(context.getSuite().getParentModule());
+      if (parentModule != null) {
+        moduleInstances.add(newParentModule(parentModule));
+      } else {
+        throw new TestNGException(
+                "Cannot load parent Guice module class: " + context.getSuite().getParentModule());
+      }
+    }
+
+    // Reuse the previous injector, if any, but don't create a child injector as JIT bindings can conflict
     Injector injector = context.getInjector(moduleInstances);
     if (injector == null) {
-      injector = parentInjector.createChildInjector(moduleInstances);
+      Stage stage;
```

Comment:
We could perhaps do something like this (and avoid the `else` part)

```java
Stage stage = Stage.DEVELOPMENT;
String stageString = context.getSuite().getGuiceStage();
if (isStringNotEmpty(stageString)){
    stage = Stage.valueOf(stageString);
}
```

Reply:
👍